src/Engines/webgpuEngine.ts

```diff
@@ -883,16 +947,94 @@ export class WebGPUEngine extends Engine {
         }
 
         // We need to recreate the render pass so that the new parameters for clear color / depth / stencil are taken into account
+        const hasScissor = this._scissorIsActive();
         if (this._currentRenderTarget) {
-            if (this._currentRenderPass) {
-                this._endRenderTargetRenderPass();
+            if (hasScissor) {
+                if (!this._rttRenderPassWrapper.renderPass) {
+                    this._startRenderTargetRenderPass(this._currentRenderTarget!, false, backBuffer ? color : null, depth, stencil);
+                }
+                this._applyScissor(this._currentRenderPass!);
+                this._clearFullQuad(backBuffer ? color : null, depth, stencil);
+            } else {
+                if (this._currentRenderPass) {
+                    this._endRenderTargetRenderPass();
+                }
+                this._startRenderTargetRenderPass(this._currentRenderTarget!, true, backBuffer ? color : null, depth, stencil);
             }
-            this._startRenderTargetRenderPass(this._currentRenderTarget!, backBuffer ? color : null, depth, stencil);
         } else {
-            this._startMainRenderPass(true, backBuffer ? color : null, depth, stencil);
+            if (!this._mainRenderPassWrapper.renderPass || !hasScissor) {
+                this._startMainRenderPass(!hasScissor, backBuffer ? color : null, depth, stencil);
+            }
+            if (hasScissor) {
+                this._applyScissor(this._currentRenderPass!);
+                this._clearFullQuad(backBuffer ? color : null, depth, stencil);
+            }
         }
     }
 
+    private _clearFullQuad(clearColor?: Nullable<IColor4Like>, clearDepth?: boolean, clearStencil?: boolean): void {

```

Comment:
As this creates GC should we merge and you do the cache in another PR ?

Reply:
Ok.